### ISSUE TITLE
Error encountered when trying to take advantage of GPU

### ISSUE THREAD

**Issue by David-Xiang** [BrainJS/brain.js#282]
![A GIF or MEME to give some spice of the internet](http://5b0988e595225.cdn.sohucs.com/images/20171101/1f63acfc41d046b8976da8717854e654.gif)

## *What* is wrong?

I am try to construct a network to train MNIST on browser. The code runs secessfully using API `brain.NeuronNetwork()`. However when I want to take advantage of GPU, I try to call `brain.NeuronNetWorkGPU()` instead and an error occurred.
How to solve this? Or do I misuse the API?

The error looks like this (in chrome devtool):
```javascript
An error occurred compiling the shaders: ERROR: 0:160: 'user_valueBitRatio' : redefinition

build @ browser.js:18105
run @ browser.js:20023
shortcut @ browser.js:16264
(anonymous) @ browser.js:876
_prepTraining @ browser.js:873
train @ browser.js:1527
train @ brainjs-gpu.js:44
async function (async)
train @ brainjs-gpu.js:41
main @ brainjs-gpu.js:90
async function (async)
main @ brainjs-gpu.js:89

browser.js:18106 Uncaught (in promise) Error: Error compiling fragment shader
    at WebGL2Kernel.build (browser.js:18106)
    at WebGL2Kernel.run (browser.js:20023)
    at NeuralNetworkGPU.shortcut [as _texturizeInputData] (browser.js:16264)
    at browser.js:876
    at Array.map (<anonymous>)
    at NeuralNetworkGPU._prepTraining (browser.js:873)
    at NeuralNetworkGPU.train (browser.js:1527)
    at train (brainjs-gpu.js:44)

build @ browser.js:18106
run @ browser.js:20023
shortcut @ browser.js:16264
(anonymous) @ browser.js:876
_prepTraining @ browser.js:873
train @ browser.js:1527
train @ brainjs-gpu.js:44
async function (async)
main @ brainjs-gpu.js:89
```

## *Where* does it happen?

MacBook Pro / Intel Core i5 / macOS 10.13.6
Chrome Version 69.0.3497.100 (Official Build) (64-bit)

## *How* do we replicate the issue?

The relevant code is:
```javascript
async function train(data){
    for (let i = 0; i < TRAIN_BATCHES; i++){
        let batch = await data.nextTrainBatch(BATCH_SIZE);
        let trainData = getStdInput(batch.xs, batch.labels);

        net.train(trainData, {
            iterations: 1,
            learningRate: LEARNING_RATE,
            momentum: 0.0001,
            log: VERBOSE,
            logPeriod: 1
        });
    }
}

async function main(){
    net = new brain.NeuralNetworkGPU({
        hiddenLayers: [512, 512],
    });
    let data = await load();
    await train(data);
}
```

**Comment by robertleeplummerjr** [comment]
Can you create a jsfiddle that reproduced the issue in the mean time while I look into this?

**Comment by David-Xiang** [comment]
https://jsfiddle.net/David_Xiang/at6bf0sc/24/
If you edit line 212 `brain.NeuralNetworkGPU` => `brain.NeuralNetwork`, you can see the normal training output in console.

**Comment by robertleeplummerjr** [comment]
Should be resolved now in v1.5.0, released yesterday.  Ty for finding!
